// chain-registry/testnet/noistestnet/assets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const info = {
    $schema: '../../assetlist.schema.json',
    chain_name: 'noistestnet',
    assets: [{
            description: 'The native token of Nois',
            denom_units: [{
                    denom: 'unois',
                    exponent: 0
                }, {
                    denom: 'nois',
                    exponent: 6
                }],
            base: 'unois',
            name: 'Nois',
            display: 'nois',
            symbol: 'NOIS',
            logo_URIs: {
                png: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/nois/images/nois.png',
                svg: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/nois/images/nois.svg'
            },
            keywords: [
                'nois',
                'randomness',
                'drand',
                'wasm'
            ],
            images: [{
                    png: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/nois/images/nois.png',
                    svg: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/nois/images/nois.svg'
                }]
        }]
};
exports.default = info;
