// chain-registry/mainnet/named.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cronos = exports.crescent = exports.coss = exports.cosmoshub = exports.coreum = exports.conscious = exports.composable = exports.commercionetwork = exports.comdex = exports.cnhostables = exports.cifer1 = exports.cifer = exports.chronicnetwork = exports.chimba = exports.chihuahua = exports.cheqd = exports.chain4energy = exports.cerberus = exports.celestia = exports.carbon = exports.canto = exports.bostrom = exports.bluzelle = exports.bluechip = exports.blockx = exports.bitsong = exports.bitcanna = exports.bitbadges = exports.beezee = exports.bandchain = exports.axelar = exports.aura1 = exports.aura = exports.assetmantle = exports.arkh = exports.archway = exports.andromeda1 = exports.andromeda = exports.althea = exports.akiro = exports.akash = exports.aioz = exports.agoric = exports.acrechain = exports._8ball = exports.tron = exports.ton = exports.solana = exports.rootstock = exports.ethereum = void 0;
exports.loyal = exports.lorenzo = exports.loop = exports.logos = exports.likecoin = exports.lava = exports.lambda = exports.kyve = exports.kujira = exports.konstellation = exports.kichain = exports.kava = exports.juno = exports.joltify = exports.jackal = exports.irisnet = exports.injective = exports.imversed = exports.impacthub = exports.idep = exports.humans = exports.highbury = exports.heli = exports.haqq = exports.gravitybridge = exports.govgen = exports.gitopia = exports.genesisl1 = exports.gateway = exports.galaxy = exports.fxcore = exports.furya = exports.firmachain = exports.finschia = exports.fetchhub = exports.evmos = exports.ethos = exports.empowerchain = exports.emoney = exports.echelon = exports.dyson = exports.dymension = exports.dydx = exports.doravota = exports.dig = exports.dhealth = exports.desmos = exports.decentr = exports.cudos = exports.cryptoorgchain = void 0;
exports.rizon = exports.regen = exports.rebus = exports.realio = exports.qwoyn = exports.quicksilver = exports.quasar = exports.qfs = exports.pylons = exports.pundix = exports.pryzm = exports.provenance = exports.point = exports.planq = exports.persistence = exports.passage1 = exports.passage = exports.panacea = exports.paloma = exports.osmosis = exports.oraichain = exports.onomy = exports.onex = exports.omniflixhub = exports.okexchain = exports.odin = exports.octa = exports.nyx = exports.nomic = exports.nolus = exports.nois = exports.noble = exports.nim = exports.nibiru = exports.neutron = exports.neutaro = exports.neura = exports.mythos = exports.mun = exports.mtgbp = exports.mises = exports.migaloo = exports.microtick = exports.meme = exports.medasdigital = exports.mayachain = exports.mars = exports.mande = exports.lumnetwork = exports.lumenx = void 0;
exports.zetachain = exports.xpla = exports.vidulum = exports.uptick = exports.ununifi = exports.unification = exports.unicorn = exports.umee = exports.titan = exports.thorchain = exports.tgrade = exports.terra2 = exports.terra = exports.terpnetwork = exports.teritori = exports.tenet = exports.taketitan = exports.synternet = exports.sunrise = exports.stride = exports.stratos = exports.starname = exports.stargaze = exports.stafihub = exports.source = exports.sommelier = exports.six = exports.sifchain = exports.shido = exports.shentu = exports.shareledger = exports.sge = exports.sentinel = exports.self = exports.sei = exports.seda = exports.secretnetwork = exports.scorum = exports.saga = exports.routerchain = void 0;
exports.ethereum = __importStar(require("./ethereum"));
exports.rootstock = __importStar(require("./rootstock"));
exports.solana = __importStar(require("./solana"));
exports.ton = __importStar(require("./ton"));
exports.tron = __importStar(require("./tron"));
exports._8ball = __importStar(require("./8ball"));
exports.acrechain = __importStar(require("./acrechain"));
exports.agoric = __importStar(require("./agoric"));
exports.aioz = __importStar(require("./aioz"));
exports.akash = __importStar(require("./akash"));
exports.akiro = __importStar(require("./akiro"));
exports.althea = __importStar(require("./althea"));
exports.andromeda = __importStar(require("./andromeda"));
exports.andromeda1 = __importStar(require("./andromeda1"));
exports.archway = __importStar(require("./archway"));
exports.arkh = __importStar(require("./arkh"));
exports.assetmantle = __importStar(require("./assetmantle"));
exports.aura = __importStar(require("./aura"));
exports.aura1 = __importStar(require("./aura1"));
exports.axelar = __importStar(require("./axelar"));
exports.bandchain = __importStar(require("./bandchain"));
exports.beezee = __importStar(require("./beezee"));
exports.bitbadges = __importStar(require("./bitbadges"));
exports.bitcanna = __importStar(require("./bitcanna"));
exports.bitsong = __importStar(require("./bitsong"));
exports.blockx = __importStar(require("./blockx"));
exports.bluechip = __importStar(require("./bluechip"));
exports.bluzelle = __importStar(require("./bluzelle"));
exports.bostrom = __importStar(require("./bostrom"));
exports.canto = __importStar(require("./canto"));
exports.carbon = __importStar(require("./carbon"));
exports.celestia = __importStar(require("./celestia"));
exports.cerberus = __importStar(require("./cerberus"));
exports.chain4energy = __importStar(require("./chain4energy"));
exports.cheqd = __importStar(require("./cheqd"));
exports.chihuahua = __importStar(require("./chihuahua"));
exports.chimba = __importStar(require("./chimba"));
exports.chronicnetwork = __importStar(require("./chronicnetwork"));
exports.cifer = __importStar(require("./cifer"));
exports.cifer1 = __importStar(require("./cifer1"));
exports.cnhostables = __importStar(require("./cnhostables"));
exports.comdex = __importStar(require("./comdex"));
exports.commercionetwork = __importStar(require("./commercionetwork"));
exports.composable = __importStar(require("./composable"));
exports.conscious = __importStar(require("./conscious"));
exports.coreum = __importStar(require("./coreum"));
exports.cosmoshub = __importStar(require("./cosmoshub"));
exports.coss = __importStar(require("./coss"));
exports.crescent = __importStar(require("./crescent"));
exports.cronos = __importStar(require("./cronos"));
exports.cryptoorgchain = __importStar(require("./cryptoorgchain"));
exports.cudos = __importStar(require("./cudos"));
exports.decentr = __importStar(require("./decentr"));
exports.desmos = __importStar(require("./desmos"));
exports.dhealth = __importStar(require("./dhealth"));
exports.dig = __importStar(require("./dig"));
exports.doravota = __importStar(require("./doravota"));
exports.dydx = __importStar(require("./dydx"));
exports.dymension = __importStar(require("./dymension"));
exports.dyson = __importStar(require("./dyson"));
exports.echelon = __importStar(require("./echelon"));
exports.emoney = __importStar(require("./emoney"));
exports.empowerchain = __importStar(require("./empowerchain"));
exports.ethos = __importStar(require("./ethos"));
exports.evmos = __importStar(require("./evmos"));
exports.fetchhub = __importStar(require("./fetchhub"));
exports.finschia = __importStar(require("./finschia"));
exports.firmachain = __importStar(require("./firmachain"));
exports.furya = __importStar(require("./furya"));
exports.fxcore = __importStar(require("./fxcore"));
exports.galaxy = __importStar(require("./galaxy"));
exports.gateway = __importStar(require("./gateway"));
exports.genesisl1 = __importStar(require("./genesisl1"));
exports.gitopia = __importStar(require("./gitopia"));
exports.govgen = __importStar(require("./govgen"));
exports.gravitybridge = __importStar(require("./gravitybridge"));
exports.haqq = __importStar(require("./haqq"));
exports.heli = __importStar(require("./heli"));
exports.highbury = __importStar(require("./highbury"));
exports.humans = __importStar(require("./humans"));
exports.idep = __importStar(require("./idep"));
exports.impacthub = __importStar(require("./impacthub"));
exports.imversed = __importStar(require("./imversed"));
exports.injective = __importStar(require("./injective"));
exports.irisnet = __importStar(require("./irisnet"));
exports.jackal = __importStar(require("./jackal"));
exports.joltify = __importStar(require("./joltify"));
exports.juno = __importStar(require("./juno"));
exports.kava = __importStar(require("./kava"));
exports.kichain = __importStar(require("./kichain"));
exports.konstellation = __importStar(require("./konstellation"));
exports.kujira = __importStar(require("./kujira"));
exports.kyve = __importStar(require("./kyve"));
exports.lambda = __importStar(require("./lambda"));
exports.lava = __importStar(require("./lava"));
exports.likecoin = __importStar(require("./likecoin"));
exports.logos = __importStar(require("./logos"));
exports.loop = __importStar(require("./loop"));
exports.lorenzo = __importStar(require("./lorenzo"));
exports.loyal = __importStar(require("./loyal"));
exports.lumenx = __importStar(require("./lumenx"));
exports.lumnetwork = __importStar(require("./lumnetwork"));
exports.mande = __importStar(require("./mande"));
exports.mars = __importStar(require("./mars"));
exports.mayachain = __importStar(require("./mayachain"));
exports.medasdigital = __importStar(require("./medasdigital"));
exports.meme = __importStar(require("./meme"));
exports.microtick = __importStar(require("./microtick"));
exports.migaloo = __importStar(require("./migaloo"));
exports.mises = __importStar(require("./mises"));
exports.mtgbp = __importStar(require("./mtgbp"));
exports.mun = __importStar(require("./mun"));
exports.mythos = __importStar(require("./mythos"));
exports.neura = __importStar(require("./neura"));
exports.neutaro = __importStar(require("./neutaro"));
exports.neutron = __importStar(require("./neutron"));
exports.nibiru = __importStar(require("./nibiru"));
exports.nim = __importStar(require("./nim"));
exports.noble = __importStar(require("./noble"));
exports.nois = __importStar(require("./nois"));
exports.nolus = __importStar(require("./nolus"));
exports.nomic = __importStar(require("./nomic"));
exports.nyx = __importStar(require("./nyx"));
exports.octa = __importStar(require("./octa"));
exports.odin = __importStar(require("./odin"));
exports.okexchain = __importStar(require("./okexchain"));
exports.omniflixhub = __importStar(require("./omniflixhub"));
exports.onex = __importStar(require("./onex"));
exports.onomy = __importStar(require("./onomy"));
exports.oraichain = __importStar(require("./oraichain"));
exports.osmosis = __importStar(require("./osmosis"));
exports.paloma = __importStar(require("./paloma"));
exports.panacea = __importStar(require("./panacea"));
exports.passage = __importStar(require("./passage"));
exports.passage1 = __importStar(require("./passage1"));
exports.persistence = __importStar(require("./persistence"));
exports.planq = __importStar(require("./planq"));
exports.point = __importStar(require("./point"));
exports.provenance = __importStar(require("./provenance"));
exports.pryzm = __importStar(require("./pryzm"));
exports.pundix = __importStar(require("./pundix"));
exports.pylons = __importStar(require("./pylons"));
exports.qfs = __importStar(require("./qfs"));
exports.quasar = __importStar(require("./quasar"));
exports.quicksilver = __importStar(require("./quicksilver"));
exports.qwoyn = __importStar(require("./qwoyn"));
exports.realio = __importStar(require("./realio"));
exports.rebus = __importStar(require("./rebus"));
exports.regen = __importStar(require("./regen"));
exports.rizon = __importStar(require("./rizon"));
exports.routerchain = __importStar(require("./routerchain"));
exports.saga = __importStar(require("./saga"));
exports.scorum = __importStar(require("./scorum"));
exports.secretnetwork = __importStar(require("./secretnetwork"));
exports.seda = __importStar(require("./seda"));
exports.sei = __importStar(require("./sei"));
exports.self = __importStar(require("./self"));
exports.sentinel = __importStar(require("./sentinel"));
exports.sge = __importStar(require("./sge"));
exports.shareledger = __importStar(require("./shareledger"));
exports.shentu = __importStar(require("./shentu"));
exports.shido = __importStar(require("./shido"));
exports.sifchain = __importStar(require("./sifchain"));
exports.six = __importStar(require("./six"));
exports.sommelier = __importStar(require("./sommelier"));
exports.source = __importStar(require("./source"));
exports.stafihub = __importStar(require("./stafihub"));
exports.stargaze = __importStar(require("./stargaze"));
exports.starname = __importStar(require("./starname"));
exports.stratos = __importStar(require("./stratos"));
exports.stride = __importStar(require("./stride"));
exports.sunrise = __importStar(require("./sunrise"));
exports.synternet = __importStar(require("./synternet"));
exports.taketitan = __importStar(require("./taketitan"));
exports.tenet = __importStar(require("./tenet"));
exports.teritori = __importStar(require("./teritori"));
exports.terpnetwork = __importStar(require("./terpnetwork"));
exports.terra = __importStar(require("./terra"));
exports.terra2 = __importStar(require("./terra2"));
exports.tgrade = __importStar(require("./tgrade"));
exports.thorchain = __importStar(require("./thorchain"));
exports.titan = __importStar(require("./titan"));
exports.umee = __importStar(require("./umee"));
exports.unicorn = __importStar(require("./unicorn"));
exports.unification = __importStar(require("./unification"));
exports.ununifi = __importStar(require("./ununifi"));
exports.uptick = __importStar(require("./uptick"));
exports.vidulum = __importStar(require("./vidulum"));
exports.xpla = __importStar(require("./xpla"));
exports.zetachain = __importStar(require("./zetachain"));
